runtime/gc_base/StringTable.cpp

```diff
@@ -567,6 +615,10 @@ j9gc_createJavaLangString(J9VMThread *vmThread, U_8 *data, UDATA length, UDATA s
 				break;
 			}
 		}
+		isASCIIorLatin1 = isASCII;
+		if (!isASCII && compressStrings && J2SE_VERSION(vm) >= J2SE_V17) {
```

Comment:
Like above, this shouldn't loop over the data twice. We can extend the above check for > 0x7F to check the UTF8 characters.

Reply:
This is a slightly different case, for UTF8 byte stream it is not straightforward to check if it is greater than `0xFF`, `isLatin1String:decodeUTF8CharN()` is required.

Reply:
Moved the code in question similarly.